google/cloud/internal/rest_carrier.cc

```diff
@@ -33,8 +33,8 @@ opentelemetry::nostd::string_view RestCarrier::Get(
 void RestCarrier::Set(opentelemetry::nostd::string_view key,
                       opentelemetry::nostd::string_view value) noexcept {
   internal::NoExceptAction([this, key, value] {
-    context_.AddHeader(std::string(key.data(), key.size()),
-                       std::string(value.data(), value.size()));
+    context_.AddDistinctHeader(std::string(key.data(), key.size()),
+                               std::string(value.data(), value.size()));
```

Comment:
I don't understand why we would do this.

[The HTTP headers spec](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2) says: "Multiple message-header fields with the same field-name MAY be present in a message if" blah blah blah...

Even if there were a propagator that sets multiple values with the same key (there isn't), we would want to add both of those values to the headers multimap.

I think you should revert the latest commit.

Reply:
I agree, but it begs the original question of why we would choose to implement `TextMapCarrier` using `RestContext`.  `std::unordered_map<std::string, std::string>` would seem to suffice.

Reply:
Ok leaving as is.

I don't understand how the HTTP Headers are used enough-- just trying to make them 1:1 with the OTel API.

Reverted commit.

Reply:
> why we would choose to implement `TextMapCarrier` using `RestContext`?

If we want to win OTel brownie points for using their intended API. It isn't necessary.

It is more important in cases that are like:
- I am a generic HTTP library
- I don't know what format to encode the trace context in
- The application sending the request knows what format to encode the trace context in
- So I just say how to write any encoding of the trace context, using that `TextMapCarrier` class